forloop_modules/globals/db_connection.py

```diff
@@ -166,4 +166,41 @@ def decrypt_db_details(database: dict) -> DbDetails:
 
         return create_db_details_from_database_dict(db_dict=database)
     else:
-        raise ValueError("Private key not found in Redis.")
+        raise ValueError(f"Private key not found for Project:{database['project_uid']} in Redis.")
```

Comment:
[Suggestion] database doesn't have to have "project_uid" key / None? - i.e. exception in exception

Reply:
This might be the case. I would keep this line nonetheless, as a simple assertion.